### ISSUE TITLE
Please learn proper version control

### ISSUE THREAD

**Issue by tycrek** [bebeli555/ElytraHighwayBot#3]
I've noticed your commits consist of deleting and re-adding files. This is **not** a good way to manage source code. Git and GitHub make it easy for you to make changes and to accept changes from others. For example, my pull request for the yaw fix could have been added with a single click. That also would have credited me in the project Contributors.

Using commits, pull requests and merging properly can drastically speed up development and improve the codebase as it is easier for others to help. [This guide](https://towardsdatascience.com/getting-started-with-git-and-github-6fcd0f2d4ac6) is a good primer on using Git.

Your Elytra bot is a fantastic tool and I'd love to help with what you already have as I have a lot experience with Git/GitHub and Java. One of the awesome things you can do with GitHub is automatically building the Jar file when you merge a commit into `master`. You should also be using the **Releases** feature to publish your Jar files rather than creating a new repository. Of course, it is your project and at the end of the day you have final say. 

I've looked through the code and I see a lot of potential changes to drastically reduce the amount of code and speed up development. However the only good way to accept these changes are with Pull Requests.

If you want to talk 1-on-1 I'd be more than happy to help over [email](mailto:josh.moore@jmoore.dev) or you can message me on Discord at **tycrek#0694** 🙂

**Comment by 5HT2** [comment]
I tried telling him this and offering to teach him on Discord and he refused ¯\\\_(ツ)_/¯

I never thought I'd see someone like you here, I'm glad I read this issue because I was *really* wondering why you of all people was contributing to something like this. 

Glad to see people like you trying to teach people, it's always great to see new contributors to open source, just hope he actually reads it :3

**Comment by Natan515** [comment]
If u want him to read ur pull req, u should dm him cus he doesn't look at it.

**Comment by Natan515** [comment]
// First check if Yaw is out of valid range (positive or negative 0-180)
		// This prevents snapping to the wrong axis on diagonals
		// Credit to tycrek for this fix.
		if (Yaw < -180.0) Yaw += 360;
		else if (Yaw > 180) Yaw -= 360; 
He did add it he just didn't accapt ur pull req.

**Comment by tycrek** [comment]
> He did add it he just didn't accapt ur pull req.

That's what I'm annoyed with because I would have been properly credited but instead he just took the fix for himself.

"Properly" credited as a Contributor to the project instead of just a comment.

Oh well, bot is useless now anyways ¯\\\_(ツ)\_/¯

**Comment by bebeli555** [comment]
Yeah. i never used git before when i started this project so it was a big mess but i think ive learned quite a bit now.
Sorry about the pull request tho i didnt really understand how they worked back then lol.

**Comment by Natan515** [comment]
> Oh well, bot is useless now anyways ¯\_(ツ)_/¯

I mean the bot might be able to use fireworks and it can save tons of time for 2b2t players.
